crates/evm/evm/src/inspectors/script.rs

```diff
@@ -14,21 +21,27 @@ pub struct ScriptExecutionInspector {
     pub script_address: Address,
 }
 
-impl<DB: Database> Inspector<DB> for ScriptExecutionInspector {
+impl<CTX, D> Inspector<CTX, EthInterpreter> for ScriptExecutionInspector
+where
+    D: Database<Error = DatabaseError>,
+    CTX: ContextTr<Db = D>,
+    CTX::Journal: JournalExt,
+{
     #[inline]
-    fn step(&mut self, interpreter: &mut Interpreter, _ecx: &mut EvmContext<DB>) {
+    fn step(&mut self, interpreter: &mut Interpreter, _ecx: &mut CTX) {
         // Check if both target and bytecode address are the same as script contract address
         // (allow calling external libraries when bytecode address is different).
-        if interpreter.current_opcode() == ADDRESS &&
-            interpreter.contract.target_address == self.script_address &&
-            interpreter.contract.bytecode_address.unwrap_or_default() == self.script_address
+        if interpreter.bytecode.opcode() == ADDRESS &&
+            interpreter.input.target_address == self.script_address
+        // TODO: Find REVM 20 equivalent
+        // && interpreter.contract.bytecode_address == self.script_address
```

Comment:
Hi @rakita, is there an equivalent of the `interpreter.contract.bytecode_address.unwrap_or_default() == self.script_address` check in the latest Revm?

Reply:
Has been implemented on Revm's side, part of the https://github.com/bluealloy/revm/compare/rakita/v21-inspector-fix branch